### ISSUE TITLE
startproject ignored if workspace is set to the same name as another project within the workspace

### ISSUE THREAD

**Issue by danieldownes** [premake/premake-core#2220]
**What seems to be the problem?**
We have a workspace called "OurEngine", with a project also titled "OurEngine" that compliles to a lib.
We have another project called "Sandbox" that compiles to a ConsoleApp.
Naturally we want the start project to be "Sandbox".

However the solution always sets the start project to "OurEngine" in the generated VS 2022 sollution.

If we change the workspace xor project name to "OurCoolEngine", so that the names are different, then the start project is set correctly.

**What did you expect to happen?**
Allow the startproject to point to a different project, if the workspace and a project whos name matches the workspace project.

**What have you tried so far?**
Describe any steps you've taken to try to solve or workaround the bug.

**How can we reproduce this?**

- [x] Visual Studio 2022 (vs2022)
- [ ] Visual Studio 2019 (vs2019)
- [ ] Visual Studio 2017 (vs2017)
- [ ] Visual Studio 2015 (vs2015)
- [ ] Visual Studio 2012 (vs2012)
- [ ] Visual Studio 2010 (vs2010)
- [ ] Visual Studio 2008 (vs2008)
- [ ] Visual Studio 2005 (vs2005)
- [ ] GNU Makefile (gmake)
- [ ] GNU Makefile 2 (gmake2)
- [ ] XCode (xcode)
- [ ] Codelite
- [ ] Other (Please list below)

**What version of Premake are you using?**
premake5 (Premake Build Script Generator) 5.0.0-beta2

**Anything else we should know?**

*(You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to issues like these!)*

**Comment by danieldownes** [comment]
It appears this may be a caching issue with VS. But I am unsure, if I set the start project manually in VS, delete all the VS files, then regenerate it does indeed retain the intended start project.

**Comment by samsinsane** [comment]
> It appears this may be a caching issue with VS.

This is correct, the cache is in the `.vs` folder beside the `.sln` file. If you delete that folder, you should see the start up project change. I'm not sure there's much we can do to resolve the issue though, I'm fairly certain those files are opened exclusively by VS and are in some binary format.
